### DISCUSSION TITLE
代理的连接不稳定，telegram无法上传文件

### DISCUSSION THREAD

**Discussion by nekonamic** [kkkgo/PaoPaoGateWay#135] [问和答] [answered]
网关使用opnsense，ppgw用的原版iso，代理通过socks5连接到本地的hysteria2，走代理的网站经常加载缓慢，通过代理下载文件隔段时间就会连接失败，telegram的cidr转发给ppgw了消息与文件可以正常发送接收但无法上传文件。请问如何解决。

**Comment by kkkgo** [comment]
可能与路由器防火墙相关规则有关，像ros有ddos规则就有这个问题（其他人的帖子有提到）
可以排查一下路由器或者尝试更换内核然后由ppgw直接连接节点。

**Reply by nekonamic** [reply]
感谢回复，经过排查断流是hysteria协议的问题，尝试使用mihomo内核 telegram仍然无法上传文件，已确定所有cidr都转发给ppgw了

**Comment by nekonamic** [comment, marked as answer]
已解决，断流是reality和hysteria2的特性，telegrm无法上传是网关防火墙规则没有设置好
